File: nos2jdbc-tutorial-wildfly/src/main/java/nos2jdbc/tutorial/wildfly/entity/Club.java
package nos2jdbc.tutorial.wildfly.entity;

import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;

@Entity
public class Club extends EntityBase {
    @Column(nullable = false, unique = true)
    public String name;
    
    @OneToMany(mappedBy = "club")
    public List<ClubMemberRel> clubMemberRelList;

}

File: nos2jdbc-tutorial-kotlin-spring/src/main/kotlin/nos2jdbc/tutorial/kotlinspring/entity/Club.kt
package nos2jdbc.tutorial.kotlinspring.entity

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Column
import jakarta.persistence.OneToMany
import java.util.ArrayList

@Entity
class Club : EntityBase() {
    @Column(nullable = false, unique = true)
    var name: String = ""

    @OneToMany(mappedBy = "club")
    var clubMemberRelList: List<ClubMemberRel> = mutableListOf() 
}

/*
@Entity
data class Club(
   @Id
    @GeneratedValue
    var id: Long = 0,
    
    @Column(nullable = false, unique = true)
    var name: String = "",
    
    @OneToMany(mappedBy = "club")
    var clubMemberRelList: List<ClubMemberRel> = mutableListOf() 
) {
	override fun hashCode(): Int {
		return id.hashCode();
	}
}
 	*/